src/components/Pages/MainPage/MainPage.tsx

```diff
@@ -15,33 +15,46 @@ import { Creations } from '../../Creations'
 import Overview from '../../Overview'
 import { PageLayout } from '../../PageLayout'
 import { ProfileInformation } from '../../ProfileInformation'
+import { Referrals } from '../../Referrals'
 import LoadingPage from '../LoadingPage'
 import { nullAddress } from './constants'
 import { Props } from './MainPage.types'
 import styles from './MainPage.module.css'
 
 function MainPage(props: Props) {
-  const { isLoading, profileAddress, loggedInAddress, isBlocked } = props
+  const { isLoading, profileAddress, loggedInAddress, isBlocked, isReferralEnabled, isLoggedIn, isLoadingFeatures } = props
```

Comment:
The concept of being logged in is used in treated in the profile app as views. Below this line, there's the `getView` function, which computes the difference between the `loggedInAddress` and the `profileAddress` which results in `OWN` if the user is logged in and is viewing their own profile and `OTHER` otherwise. Would you mind using that concept?